lib/hash-tree.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+var crypto = require('crypto');
+var fs = require('fs');
+
+function getFileInfos(fullPath) {
+  var stat = fs.statSync(fullPath);
+
+  if (stat.isFile()) {
+    return [{
+      fullPath: fullPath,
+      mtime: stat.mtime.getTime(),
+      mode: stat.mode,
+      size: stat.size
+    }];
+  } else if (stat.isDirectory()) {
+    // if it ends with node_modules do nothing
+    return fs.readdirSync(fullPath).sort().reduce(function(paths, entry) {
+      if (entry.toLowerCase() === 'node_modules') {
+        return paths;
+      }
+
+      return paths.concat(getFileInfos(fullPath + '/' + entry));
+    }, []);
+  } else {
+    throw new Error('"' + fullPath + '": Unexpected file type');
+  }
+}
+
+function stringifyFileInfo(fileInfo) {
+  return '\x00' + fileInfo.mtime + '\x00' + fileInfo.mode + '\x00' + fileInfo.size;
```

Comment:
I'm not sure if '0x00' is what you want, as it just puts a '0' into the string. Why not something that won't generally show up in a path, like '|'?

Reply:
i was mimicking the old code from broccoli-kitchen-sink-helpers. `|` is good, less funky looking